tests/e2e/improved_csi_idempotency.go

```diff
@@ -344,6 +514,55 @@ func createVolumeWithServiceDown(serviceName string, namespace string, client cl
 
 		ginkgo.By(fmt.Sprintf("Sleeping for %v seconds to allow full sync finish", fullSyncWaitTime))
 		time.Sleep(time.Duration(fullSyncWaitTime) * time.Second)
+	} else if serviceName == hostdServiceName {
+		ginkgo.By("Fetch IPs for the all the hosts in the cluster")
+		hostIPs := e2eVSphere.getAllHostsIP(ctx)
+		isServiceStopped = true
+		for index, hostIP := range hostIPs {
+			// Safe desctructive test: Stop hostd service only on half number of ESX Hosts
+			if index%2 == 0 {
+				ginkgo.By(fmt.Sprintf("Stopping hostd service on the host  %s ...", hostIP))
+				_, err := runHostDCommandOnHost(hostIP, stopOperation)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				op, err := runHostDCommandOnHost(hostIP, statusOperation)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				gomega.Expect(strings.Contains(op, "hostd is not running.")).NotTo(gomega.BeFalse())
+			}
```

Comment:
I had earlier meant to put this whole thing into a separate method

Reply:
Refactored the code again.